shim.c

```diff
@@ -1468,43 +1557,54 @@ load_certs(EFI_HANDLE image_handle)
 		goto done;
 	}
 
-	while (1) {
-		UINTN old = buffersize;
-		efi_status = dir->Read(dir, &buffersize, buffer);
-		if (efi_status == EFI_BUFFER_TOO_SMALL) {
-			if (buffersize == old) {
-				/*
-				 * Some UEFI drivers or firmwares are not compliant with
-				 * the EFI_FILE_PROTOCOL.Read() specs and do not return the
-				 * required buffer size along with EFI_BUFFER_TOO_SMALL.
-				 * Work around this by progressively increasing the buffer
-				 * size, up to a certain point, until the call succeeds.
-				 */
-				perror(L"Error reading directory %s - non-compliant UEFI driver or firmware!\n",
-				       PathName);
-				buffersize = (buffersize < 4) ? 4 : buffersize * 2;
-				if (buffersize > 1024)
+	sbat_status = load_revocations_file(image_handle, PathName);
+
+	if (secure_mode()) {
```

Comment:
scan-build (the clang analyzer) has correctly noticed that in the case where secure_mode() == false, this is a dead store.  That has me wondering why we're even doing it in disabled mode...

Reply:
This is intentional, but is worth questioning, The idea is that if there's a revocation file there that has an automatic update that it is applied no matter what. I wrote the above code before I made the change to always clear the variable with SB disabled. With the always clear behavior, I agree that this doesn't feel right anymore, and would be fine with moving it into the conditional.